src/java.base/share/classes/jdk/internal/classfile/constantpool/ConstantPoolBuilder.java

```diff
@@ -63,29 +63,22 @@
         permits SplitConstantPool, TemporaryConstantPool {
 
     /**
-     * {@return a new constant pool builder}  The new constant pool builder
-     * will inherit the classfile processing options of the specified class.
-     * If the processing options include {@link Classfile.Option#constantPoolSharing(boolean)},
-     * (the default) the new constant pool builder will be also be pre-populated with the
-     * contents of the constant pool associated with the class reader.
+     * {@return a new constant pool builder}  The new constant pool builder will
+     * be pre-populated with the contents of the constant pool associated with
+     * the class reader.
      *
      * @param classModel the class to copy from
      */
     static ConstantPoolBuilder of(ClassModel classModel) {
-        ClassReaderImpl reader = (ClassReaderImpl) classModel.constantPool();
-        return reader.options().cpSharing
-          ? new SplitConstantPool(reader)
-          : new SplitConstantPool(reader.options());
+        return new SplitConstantPool((ClassReaderImpl) classModel.constantPool());
     }
 
     /**
      * {@return a new constant pool builder}  The new constant pool builder
-     * will be empty and have the specified classfile processing options.
-     *
-     * @param options the processing options
+     * will be empty.
      */
-    static ConstantPoolBuilder of(Collection<Classfile.Option> options) {
-        return new SplitConstantPool(new Options(options));
+    static ConstantPoolBuilder of() {
+        return new SplitConstantPool();
     }
```

Comment:
The old ctors here were (ClassModel) and (Options), where the ClassModel version had a trail of breadcrumbs back to options.  The new ctors are (ClassModel) and no-args, which is a slight departure.  What use did CPB make of options in the first place, if any?